php/class-customize-snapshot-manager.php

```diff
@@ -914,4 +973,34 @@ public function get_front_uuid_param() {
 	public function get_customize_uuid_param() {
 		return constant( get_class( $this->post_type ) . '::CUSTOMIZE_UUID_PARAM_NAME' );
 	}
+
+	/**
+	 * Publishes changeset from frontend.
+	 */
+	public function ajax_snapshot_frontend_publish() {
+		if ( ! check_ajax_referer( 'customize-snapshots-frontend-publish', 'nonce' ) ) {
```

Comment:
Also important to check whether `current_user_can( get_post_type_object( 'customize_changeset' )->cap->publish_posts )`